Valigator.Text.Json/ValigatorJsonPropertyHandler.cs

```diff
@@ -18,12 +18,84 @@ internal abstract class ValigatorJsonPropertyHandler<TObject>
 
 		public static ValigatorJsonPropertyHandler<TObject> Create(PropertyInfo property)
 		{
-			var handlerType = typeof(ValigatorJsonPropertyHandler<,>).MakeGenericType(typeof(TObject), property.PropertyType.GetGenericArguments()[0]);
+			var propertyHandlerType = property.PropertyType.IsValigatorDataType() ? typeof(ValigatorJsonPropertyHandler<,>) : typeof(NonValigatorDataJsonPropertyHandler<,>);
+
+			var propertyType = property.PropertyType.IsValigatorDataType() ? property.PropertyType.GetGenericArguments()[0] : property.PropertyType;
```

Comment:
can dry this out by extracting the logic into two seperate methods that return a create method depending on IsValigatorDataType() instead of checking twice for IsValigatorDataType(

Reply:
Sprouted out.